pkg/operator/ceph/cluster/controller.go

```diff
@@ -259,17 +264,68 @@ func (r *ReconcileCephCluster) reconcile(request reconcile.Request) (reconcile.R
 		return reconcile.Result{}, cephCluster, errors.Wrap(err, "failed to add finalizer")
 	}
 
+	// After updating the status of the CephCluster, we need to make sure we have the latest
+	refreshCephCluster := func() error {
+		cephCluster, err = r.context.RookClientset.CephV1().CephClusters(cephCluster.Namespace).Get(context.TODO(), cephCluster.Name, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "failed to get latest CephCluster %q changes", request.NamespacedName.String())
+		}
+		return nil
+	}
+
 	// DELETE: the CR was deleted
 	if !cephCluster.GetDeletionTimestamp().IsZero() {
+		// Set the deleting status
+		opcontroller.UpdateClusterCondition(r.context, cephCluster, request.NamespacedName,
```

Comment:
Do we need the deleting condition? We're immediately going to be adding another event or condition if the deletion is blocked, and if it is not blocked, the whole CR will go away anyway.

Reply:
I think so. Similarly to how Kubernetes changes state to `Terminating` immediately, I think we need to set the `Deleting` phase/state so admins know that Rook is attempting to terminate the CephCluster. If it can't terminate, the `Deleting` phase/state is kept even after the failure event.